cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -641,6 +656,9 @@ def _add_fsx_storage(self, id: str, shared_fsx: SharedFsx):
                 security_group_ids=self._get_compute_security_groups(),
             )
             fsx_id = fsx_resource.ref
+            self.shared_storage_attributes[shared_fsx.shared_storage_type][
```

Comment:
If you have this code, what about adding the set of DNSName here and remove the loop below?